src/app/dashboard/layout.tsx

```diff
@@ -23,6 +23,7 @@ const queryClient = new QueryClient({
       refetchOnReconnect: false,
       refetchOnWindowFocus: false,
       gcTime: 300000,
+      placeholderData: keepPreviousData,
```

Comment:
keepPrevious data by default as we want to avoid flickering of UI. Once we implement skeleton UIs it won't be necessary for some of the components but it's a good default for now.